pennylane/plugins/default_qubit.py

```diff
@@ -218,10 +218,36 @@ def reset(self):
         self._state[0] = 1
         self._pre_rotated_state = self._state
 
-    def probability(self, wires=None):
+    def _analytic_probability(self, wires=None):
+        """Return the (marginal) analytic probability of each computational basis."""
         if self._state is None:
             return None
 
         wires = wires or range(self.num_wires)
+
         prob = self.marginal_prob(np.abs(self._state) ** 2, wires)
         return prob
+
+    def probability(self, wires=None):
+        """Return the (marginal) probability of each computational basis."""
```

Comment:
As this method is defined in the parent abstract class, no need to define the docstring here 🙂 The docstring will automatically be inherited by the method.

This has several advantages:

* Reduces maintenance overhead --- docstring needs only be updated in one place

* Helps enforce the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle), and avoids the temptation to modify the signature/docstring.

In limited cases, however, if the plugin _needs_ to add/change information in the docstring, it is okay to override to add this.

Reply:
```suggestion
```